src/recap

```diff
@@ -146,492 +134,440 @@ print_usage() {
+    # of characters.  Now we can compare the canonicalized path of fd to the
+    # tmpdir/ib* pattern:
+    [[ $(readlink -f "${name}") == "${TMPDIR}"/ib* ]] &&
+    # If any files match that pattern, we see if the first line in the file
+    # starts with "===", which is the case in the particular InnoDB file we
+    # care about:
+      head -c8 "${name}" | grep -q '^====' &&
+    # Finally, print out the information and append it to the mysql file:
+    cat "${name}"  >> "${LOGFILE}"
+  done
 }
 
 # print the output of "mysqladmin processlist" to the mysql file
 print_mysql_procs() {
-        # We only support a single DOTMYDOTCNF for MySQL when running Plesk
-        if [ -f /etc/psa/.psa.shadow ]
-        then
-                echo "MySQL processes" >> $MYSQL_FILE
-                if [ $MYSQL_PROCESS_LIST == "table" ]
-                then
-                        mysqladmin -uadmin -p`cat /etc/psa/.psa.shadow` -v processlist >> $MYSQL_FILE
-                fi
-                if [ $MYSQL_PROCESS_LIST == "vertical" ]
-                then
-                        mysql -uadmin -p`cat /etc/psa/.psa.shadow` -e "show full processlist\G" >> $MYSQL_FILE
-                fi
-                print_blankline $MYSQL_FILE
-        else
-                for cnf in $( sed -e 's/,/ /g' <<<"${DOTMYDOTCNF}" )
-                do
-                        if [ -f $cnf ]
-                        then
-                                echo "MySQL $cnf processes" >> $MYSQL_FILE
-                                if [ $MYSQL_PROCESS_LIST == "table" ]
-                                then
-                                  mysqladmin --defaults-file=$cnf -v processlist >> $MYSQL_FILE
-                                fi
-                                if [ $MYSQL_PROCESS_LIST == "vertical" ]
-                                then
-                                	# This doesnt seem to work right
-                                  mysqladmin --defaults-file=$cnf --vertical processlist
-                                fi
-                                print_blankline $MYSQL_FILE
-                        else
-                                echo "MySQL $cnf for processes does not exist. Please check DOTMYDOTCNF in /etc/recap" >> $MYSQL_FILE
-                                print_blankline $MYSQL_FILE
-                        fi
-                done
+  local LOGFILE=$1
+  # We only support a single DOTMYDOTCNF for MySQL when running Plesk
+  local PLESK_FILE="/etc/psa/.psa.shadow"
+  if [[ -r "${PLESK_FILE}" ]]; then
+    echo "MySQL processes" >> "${LOGFILE}"
+    if [[ ${MYSQL_PROCESS_LIST} == "table" ]]; then
+      mysqladmin -uadmin -p$( cat "${PLEASK_FILE}" ) -v processlist >> "${LOGFILE}"
+    fi
+    if [[ ${MYSQL_PROCESS_LIST} == "vertical" ]]; then
+      mysql -uadmin -p$( cat "${PLESK_FILE}" ) -e "show full processlist\G" >> "${LOGFILE}"
+    fi
+    print_blankline "${LOGFILE}"
+  else
+    for cnf in $( sed -e 's/,/ /g' <<<"${DOTMYDOTCNF}" ); do
```

Comment:
should be similar to https://github.com/rackerlabs/recap/pull/78#discussion_r105439236